Library/Homebrew/version.rb

```diff
@@ -171,6 +171,41 @@ def detected_from_url?
     end
   end
 
+  class NullVersion
+    include Comparable
+
+    def <=>(other)
+      -1
+    end
+
+    def eql?(other)
+      # Makes sure that the same instance of Version::NULL
+      # will never equal itself; normally Comparable#==
+      # will return true for this regardless of the return
+      # value of #<=>
+      false
```

Comment:
This mirrors `NaN` - it reflects a value that's not a number. (And it's consistent with the `#to_f` value returned by this class.)